server/apps/context.go

```diff
@@ -0,0 +1,115 @@
+package apps
+
+import (
+	"github.com/mattermost/mattermost-plugin-apps/server/store"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type Context struct {
+	AppID        store.AppID       `json:"-"`
+	ActingUserID string            `json:"acting_user_id,omitempty"`
+	UserID       string            `json:"user_id,omitempty"`
+	TeamID       string            `json:"team_id"`
+	ChannelID    string            `json:"channel_id,omitempty"`
+	PostID       string            `json:"post_id,omitempty"`
+	RootPostID   string            `json:"root_post_id,omitempty"`
+	Props        map[string]string `json:"props,omitempty"`
+	expandedContext
+}
+
+type expandedContext struct {
+	ActingUser *model.User       `json:"acting_user,omitempty"`
+	App        *store.App        `json:"app,omitempty"`
+	Channel    *model.Channel    `json:"channel,omitempty"`
+	Config     *MattermostConfig `json:"config,omitempty"`
+	Mentioned  []*model.User     `json:"mentioned,omitempty"`
+	Post       *model.Post       `json:"post,omitempty"`
+	RootPost   *model.Post       `json:"root_post,omitempty"`
+	Team       *model.Team       `json:"team,omitempty"`
+	User       *model.User       `json:"user,omitempty"`
+}
+
+type MattermostConfig struct {
+	SiteURL string `json:"site_url"`
+}
+
+type Thread struct {
+	ChannelID  string `json:"channel_id"`
+	RootPostID string `json:"root_post_id"`
+}
+
+func NewChannelContext(ch *model.Channel) *Context {
```

Comment:
I wonder how useful will be these functions, since the context will come mainly from the client, so it will send only the ids, not the whole objects.

Reply:
These are useful in notifications, the clients (should? must?) create the contexts for user-initiated actions.